tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -100,15 +92,15 @@ uint32_t DeclResultIdMapper::getDeclResultId(const NamedDecl *decl) {
 uint32_t DeclResultIdMapper::createFnParam(uint32_t paramType,
                                            const ParmVarDecl *param) {
   const uint32_t id = theBuilder.addFnParam(paramType, param->getName());
-  astDecls[param] = {id, spv::StorageClass::Function};
+  astDecls[param] = {id, spv::StorageClass::Function, -1};
```

Comment:
I would create a constructor for `DeclSpirvInfo` which defaults the `indexInCTBuffer` to `-1` in order to avoid having to manually add `-1` to all existing usages.
When I saw this line, my first reaction was: why `-1`? seemed like a random number.